include/mapbox/variant.hpp

```diff
@@ -84,25 +84,27 @@ struct MAPBOX_VARIANT_DEPRECATED static_visitor
     ~static_visitor() {}
 };
 
+using type_index_t = unsigned int;
```

Comment:
`llvm/libc++` `<variant>` implementation uses `unsigned`.  
> Shouldn't a uint8_t be good enough as outlined in the ticket? 

We probably can but I favour something along `BOOST_OPTIMIZE_SIZE` compile flag to choose min suitable `type_index_t` at compile time. 

>I think, for up to 256 alternative types variant's discriminatior type may be clearly expressed in terms of uint_fast8_t if speed is priority and uint_least8_t otherwise.

yep, sounds like an idea ^ 

/cc @daniel-j-h  @tomilov